### PR TITLE
RUN-3005: support new format of API policy definitions

### PR DESCRIPTION
**Pull request by wenjunche**
[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a0236271573627ee4384296)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a0235a81573627ee4384295)

### CODE REVIEW

File: src/modules.d.ts [hunk 1]
```diff
@@ -112,4 +116,8 @@ declare module 'electron' {
         export function readHtml(type?: string): string;
         export function readText(type?: string): string;
     }
+
+    export namespace net {
```
Comment (HarsimranSingh) [inline src/modules.d.ts]: Already defined in this file https://github.com/HadoukenIO/core/blob/develop/src/modules.d.ts#L36

File: src/browser/api_protocol/api_handlers/api_policy_processor.ts [hunk 2]
```diff
@@ -290,6 +327,28 @@ function registerDelegate(apiPath: ApiPath, delegate: ApiPolicyDelegate) {
     delegateMap.set(apiPath, delegate);
 }
 
+function retrieveAPIPolicyContent(): Promise<any> {
+    writeToLog(1, 'retrieveAPIPolicyContent', true);
+    return new Promise((resolve, reject) => {
+        const msg: GetDesktopOwnerSettings = {
+            topic: 'application',
+            action: 'get-desktop-owner-settings',
+            sourceUrl: 'https://openfin.co', // ignored by RVM is isGlobal is true
```
Comment (HarsimranSingh) [inline src/browser/api_protocol/api_handlers/api_policy_processor.ts]: :neckbeard: ignored by RVM ~~is~~ if isGlobal is true

File: src/browser/api_protocol/api_handlers/api_policy_processor.ts [hunk 3]
```diff
@@ -257,12 +286,21 @@ function requestAppPermissions(configUrl: string): Promise<any> {
             writeToLog(1, 'requestAppPermissions cached', true);
             resolve(configUrlPermissionsMap[configUrl]);
         } else {
+            const policy = searchPolicyByConfigUrl(configUrl);
```
Comment (HarsimranSingh) [inline src/browser/api_protocol/api_handlers/api_policy_processor.ts]: Fix spacing for this code section

File: src/browser/api_protocol/api_handlers/api_policy_processor.ts [hunk 4]
```diff
@@ -257,12 +286,21 @@ function requestAppPermissions(configUrl: string): Promise<any> {
...
+            if (policy && policy.permissions) {
+                    configUrlPermissionsMap[configUrl] = {permissions: policy.permissions};
+                } else {
+                    configUrlPermissionsMap[configUrl] = {};
+                }
+                resolve(configUrlPermissionsMap[configUrl]);
+            }
+
             const msg: GetDesktopOwnerSettings =  {
```
Comment (HarsimranSingh) [inline src/browser/api_protocol/api_handlers/api_policy_processor.ts]: Comment this out or remove

File: src/browser/api_protocol/api_handlers/api_policy_processor.ts [hunk 5]
```diff
@@ -257,12 +286,21 @@ function requestAppPermissions(configUrl: string): Promise<any> {
...
                 topic: 'application',
                 action: 'get-desktop-owner-settings',
                 sourceUrl: configUrl
             };
-
+            /*
```
Comment (HarsimranSingh) [inline src/browser/api_protocol/api_handlers/api_policy_processor.ts]: Can this be removed?

**Comment by whyn07m3** [comment]
Not sure, but can a test(s) be written to test this change?

**Comment by wenjunche** [comment]
Updated.

Tests:
[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a130a419a4b9e06a9723a4e)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a1304309a4b9e06a9723a4d)

Anthony created  a manual test for this change.

File: src/browser/api_protocol/api_handlers/api_policy_processor.ts [hunk 6]
```diff
@@ -12,6 +12,7 @@ import {ApiPath, ApiPolicyDelegate, Endpoint} from '../shapes';
 const rvmBus = require('../../rvm/rvm_message_bus').rvmMessageBus;  // retrieve permission setting from registry
 import { GetDesktopOwnerSettings } from '../../rvm/rvm_message_bus';
 import { writeToLog } from '../../log';
+import {app as electronApp, net} from 'electron';
```
Comment (HarsimranSingh) [inline src/browser/api_protocol/api_handlers/api_policy_processor.ts]: Why is `net` required here?

**Pull request merged**